src/adapter/src/optimize/view.rs

```diff
@@ -78,6 +78,8 @@ impl Optimize<HirRelationExpr> for Optimizer {
         // MIR ⇒ MIR optimization (local)
         let expr = if expr.as_const().is_some() {
             // No need to optimize further, because we already have a constant.
+            // But trace this at "local", so that `EXPLAIN LOCALLY OPTIMIZED PLAN` can pick it up.
+            trace_plan!(at: "local", &expr);
```

Comment:
I don't know much about the `trace_plan!` macro, but if it's okay to run at this location then LGTM!

Reply:
Yes, it's intended to be run at any moment where we need to emit an extra plan tracing event. Usually, plan tracing events are emitted by `mz_ore::instrument` annotations on functions ([example](https://github.com/MaterializeInc/materialize/blob/7893991ee11dd9601bce5592e0c73bcb1f42f365/src/transform/src/predicate_pushdown.rs#L124)), but sometimes we need to emit some extra ones, in which case we use `trace_plan!`.

(Plan tracing events are used by EXPLAIN, which usually just picks out one tracing event, which corresponds to the stage requested by the EXPLAIN statement, e.g., EXPLAIN LOCALLY OPTIMIZED PLAN picks out "local". Plan tracing events are also used by https://optimizer-trace.dev.materialize.com/ , which shows all the plan tracing events in a tree structure.)